src/Testcontainers.Azurite/AzuriteBuilder.cs

```diff
@@ -0,0 +1,481 @@
+      .WithServices(AzuriteServices.All)
+      .WithBlobPort(DefaultBlobPort)
+      .WithQueuePort(DefaultQueuePort)
+      .WithTablePort(DefaultTablePort)
+      .WithWorkspaceLocation(DefaultWorkspaceDirectoryPath);
+  }
+
+  /// <inheritdoc />
+  protected override AzuriteBuilder Clone(IResourceConfiguration<CreateContainerParameters> resourceConfiguration)
+  {
+    return Merge(DockerResourceConfiguration, new AzuriteConfiguration(resourceConfiguration));
+  }
+
+  /// <inheritdoc />
+  protected override AzuriteBuilder Clone(IContainerConfiguration resourceConfiguration)
+  {
+    return Merge(DockerResourceConfiguration, new AzuriteConfiguration(resourceConfiguration));
+  }
+
+  /// <inheritdoc />
+  protected override AzuriteBuilder Merge(AzuriteConfiguration oldValue, AzuriteConfiguration newValue)
+  {
+    return new AzuriteBuilder(new AzuriteConfiguration(oldValue, newValue));
+  }
+
+  private AzuriteBuilder AddBlobPortAndBinding()
+  {
+    if (!AllServicesEnabled && !BlobServiceOnlyEnabled)
+    {
+      return this;
+    }
+
+    var exposedPort = DockerResourceConfiguration.BlobPort ?? DefaultBlobPort;
+    var bindedPort = DockerResourceConfiguration.BlobPortBinding ?? 0;
+    return ExposeAndBindPort(exposedPort, bindedPort);
+  }
+
+  private AzuriteBuilder AddQueuePortAndBinding()
+  {
+    if (!AllServicesEnabled && !QueueServiceOnlyEnabled)
+    {
+      return this;
+    }
+
+    var exposedPort = DockerResourceConfiguration.QueuePort ?? DefaultQueuePort;
+    var bindedPort = DockerResourceConfiguration.QueuePortBinding ?? 0;
+    return ExposeAndBindPort(exposedPort, bindedPort);
+  }
+
+  private AzuriteBuilder AddTablePortAndBinding()
+  {
+    if (!AllServicesEnabled && !TableServiceOnlyEnabled)
+    {
+      return this;
+    }
+
+    var exposedPort = DockerResourceConfiguration.TablePort ?? DefaultTablePort;
+    var bindedPort = DockerResourceConfiguration.TablePortBinding ?? 0;
+    return ExposeAndBindPort(exposedPort, bindedPort);
+  }
+
+  private AzuriteBuilder AddWorkspaceBinding()
+  {
+    if (string.IsNullOrEmpty(DockerResourceConfiguration.WorkspaceLocationBinding))
+    {
+      return this;
+    }
+
+    var location = string.IsNullOrEmpty(DockerResourceConfiguration.WorkspaceLocation)
+      ? DefaultWorkspaceDirectoryPath
+      : DockerResourceConfiguration.WorkspaceLocation;
+
+    return WithBindMount(DockerResourceConfiguration.WorkspaceLocationBinding, location);
+  }
+
+  private string GetExecutable()
+  {
+    if (BlobServiceOnlyEnabled)
+    {
+      return "azurite-blob";
+    }
+
+    if (QueueServiceOnlyEnabled)
+    {
+      return "azurite-queue";
+    }
+
+    if (TableServiceOnlyEnabled)
+    {
+      return "azurite-table";
+    }
+
+    return "azurite";
+  }
+
+  private string[] GetEnabledServices()
+  {
+    const string defaultRemoteEndpoint = "0.0.0.0";
+
+    IList<string> args = new List<string>();
+
+    if (AllServicesEnabled || BlobServiceOnlyEnabled)
```

Comment:
This is good idea because there is XOR relationship between services: ALL xor BLOB xor QUEUE xor TABLE. we can provide 4 independent containers maybe with one generic builder. Not sure right now. I am just thinking out loud.